docs/src/app/(public)/(content)/react/handbook/composition/page.mdx

```diff
@@ -0,0 +1,82 @@
+# Composition
+
+<Subtitle>A guide to composing Base UI components with your own React components.</Subtitle>
+<Meta
+  name="description"
+  content="A guide to composing Base UI components with your own React components."
+/>
+
+### Composing custom React components
+
+Use the `render` prop to compose a Base UI part with your own React components.
+
+For example, most triggers render a `<button>` by default. The code snippet below shows how to use a custom button instead.
+
+```tsx title="index.tsx"
+<Menu.Trigger render={<MyButton size="md">Open menu</MyButton>} />
```

Comment:
Looks like my comment was lost (GH is having some problems)
We should recommend placing children in the original component not the one in the render prop for simplicity (similarily to Ariakit: https://ariakit.org/guide/composition#merging-the-rendered-element-props, Flat is better than nested)

Reply:
OK @colmtuite don't listen to me, listen to @michaldudak 😅